kubernetes_platform/python/kfp/kubernetes/config_map.py

```diff
@@ -45,10 +45,14 @@ def use_config_map_as_env(
             env_var=env_var,
         ) for config_map_key, env_var in config_map_key_to_env.items()
     ]
-    config_map_as_env = pb.ConfigMapAsEnv(
-        config_map_name=config_map_name,
-        key_to_env=key_to_env,
-    )
+    config_map_as_env = pb.ConfigMapAsEnv(key_to_env=key_to_env)
+
+    config_map_name_parameter = common.parse_k8s_parameter_input(config_map_name, task)
+    config_map_as_env.config_name_parameter.CopyFrom(config_map_name_parameter)
```

Comment:
Does this cause new content to be added to the output YAML if the user is using the legacy `config_map_name` since it's assigning an empty value by default?

Reply:
> since it's assigning an empty value by default?

not sure I'm following this one, can you clarify?

Reply:
I misread the code, but I guess I'm wondering if it makes sense to set `config_name_parameter` if the type is a string rather than just fallback on the deprecated field only since it's always set below.

Reply:
well since it is deprecated, we should always include the new field, in my mind we should always include the non-deprecated version

Reply:
it will for 2.4+